### PR TITLE
various tox improvements

### PR DESCRIPTION
**Pull request by gaborbernat**
- pass proxy environment variables to docs (as these are needed to fetch correctly intersphinx files behind proxies)
- better indentation for commands to see where boundaries between commands are
- do not start all definitions on new line, this makes the file more compact and readable
- use references instead of repeating dependencies, and version qualify dependency
- replace echo with Python for not relying on OS packages
- also ignore intermediate coverage files

### CODE REVIEW

**Comment by gaborbernat** [comment]
@agjohnson which ones do you mean? and what do you consider arbitrary?

**Comment by gaborbernat** [comment]
@agjohnson please clarify

**Comment by gaborbernat** [comment]
Anyone? :+1:

File: tox.ini [hunk 1]
```diff
@@ -1,56 +1,44 @@
...
-minversion=2.9.0
+minversion = 2.9.0
 envlist = py{27,36},lint,docs
 skipsdist = True
 
 [travis]
-python =
-    2.7: py27
-    3.6: py36
+python = 2.7: py27
```
Comment (stsewd) [inline tox.ini]: I think this formattings are the requested changes

File: tox.ini [hunk 2]
```diff
@@ -1,56 +1,44 @@
...
+         3.6: py36
 
 [testenv]
 description = run test suite for the application with {basepython}
-setenv =
-    PYTHONPATH={toxinidir}/readthedocs:{toxinidir}
-    DJANGO_SETTINGS_MODULE=readthedocs.settings.test
-    LANG=C
-    DJANGO_SETTINGS_SKIP_LOCAL=True
+setenv = PYTHONPATH={toxinidir}/readthedocs:{toxinidir}
```
Comment (stsewd) [inline tox.ini]: And those

File: tox.ini [hunk 3]
```diff
@@ -1,56 +1,44 @@
...
-    py.test {posargs}
+commands = pytest {posargs}
 
 [testenv:docs]
 description = build readthedocs documentation
 changedir = {toxinidir}/docs
-commands =
-    sphinx-build -b html -d {envtmpdir}/doctrees .  {envtmpdir}/html
+passenv = HOME http_proxy https_proxy no_proxy
+commands = sphinx-build -b html -d {envtmpdir}/doctrees . {envtmpdir}/html
```
Comment (stsewd) [inline tox.ini]: By arbitrary, I kind of think this is what Anthony means, maybe? (Always start the commands on a new line)

**Comment by RichardLitt** [comment]
@gaborbernat We'll get to this when we can. This is an open source project, and our priority list sometimes means there is a delay in response. Don't worry; we know this PR is here, and we'll review it again when we can.

File: tox.ini [hunk 4]
```diff
@@ -1,56 +1,44 @@
...
-    --profile=prospector-more \
-    --die-on-tool-error {posargs}
-    prospector \
-    --profile-path={toxinidir} \
-    --profile=prospector \
-    --die-on-tool-error {posargs}
+commands = prospector --profile-path={toxinidir} --die-on-tool-error \
+                --profile=prospector-more  {posargs}
+           prospector --profile-path={toxinidir} --die-on-tool-error \
+                --profile=prospector  {posargs}
```
Comment (agjohnson) [inline tox.ini]: Line continuation to revert here as well.

File: tox.ini [hunk 5]
```diff
@@ -1,56 +1,44 @@
...
-    pytest-cov
-whitelist_externals = echo
-commands =
-    py.test --disable-pytest-warnings \
-        --cov-report=term --cov-report=html --cov=. {posargs}
-    echo Annotated HTML coverage report is in {toxinidir}/readthedocs/htmlcov/index.html
+deps = {[testenv]deps}
+       pytest-cov >= 2.5.1, <3
+commands = pytest --disable-pytest-warnings --cov-report=term --cov-report=html --cov=. {posargs}
+           python -c 'print("Annotated HTML coverage report is in {toxinidir}/readthedocs/htmlcov/index.html")'
```
Comment (agjohnson) [inline tox.ini]: Line continuation to revert here as well. No need to change echo -> python either, really.

**Comment by ericholscher** [comment]
Sounds like we have different views on what should go in `tox.ini`. Thanks for the PR, but I'm going to close this as I don't think it's likely to get updated/merged. Feel free to reopen with changes if so -- sorry!

**Pull request closed without merging**